libvips/foreign/vips2jpeg.c

```diff
@@ -797,12 +806,18 @@ vips__jpeg_write_target( VipsImage *in, VipsTarget *target,
 	 */
         target_dest( &write->cinfo, target );
 
+    /* Retain old behavior for now
+     */
+    if( no_subsample ) {
+        subsample_mode = VIPS_FOREIGN_JPEG_SUBSAMPLE_OFF;
+    }
+
```

Comment:
This `if()` can go if we have the extra logic in `_build()`.

Reply:
done